packages/editor-ui/src/modules/workflows.ts

```diff
@@ -1,24 +1,25 @@
-import { getNewWorkflow } from '@/api/workflows';
+import { getNewWorkflow, getWorkflowTemplate } from '@/api/workflows';
 import { DUPLICATE_POSTFFIX, MAX_WORKFLOW_NAME_LENGTH, DEFAULT_NEW_WORKFLOW_NAME } from '@/constants';
 import { ActionContext, Module } from 'vuex';
 import {
 	IRootState,
 	IWorkflowsState,
+	IWorkflowTemplate,
 } from '../Interface';
 
 const module: Module<IWorkflowsState, IRootState> = {
 	namespaced: true,
 	state: {},
 	actions: {
-		setNewWorkflowName: async (context: ActionContext<IWorkflowsState, IRootState>): Promise<void> => {
+		setNewWorkflowName: async (context: ActionContext<IWorkflowsState, IRootState>, name?: string): Promise<void> => {
 			let newName = '';
 			try {
-				const newWorkflow = await getNewWorkflow(context.rootGetters.getRestApiContext);
+				const newWorkflow = await getNewWorkflow(context.rootGetters.getRestApiContext, name);
 				newName = newWorkflow.name;
 			}
 			catch (e) {
 				// in case of error, default to original name
-				newName = DEFAULT_NEW_WORKFLOW_NAME;
+				newName = name? name: DEFAULT_NEW_WORKFLOW_NAME;
```

Comment:
adjust spacing, so it's clearer that this is a ternary and not a typescript typing
```suggestion
				newName = name ? name : DEFAULT_NEW_WORKFLOW_NAME;
```

another question: doesn't this make a potential issue with the database if the name is already taken, but the api call errors?

Reply:
this is only returning a value clientside, user would get an error if they don't update it..

Reply:
it's just that this fallback behavior is important so that the workflow name is not empty when user opens a new workflow or loads a template